testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeTablePropertiesCompatibility.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.deltalake;
+
+import io.trino.tempto.assertions.QueryAssert;
+import io.trino.testng.services.Flaky;
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_OSS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.DATABRICKS_COMMUNICATION_FAILURE_ISSUE;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.DATABRICKS_COMMUNICATION_FAILURE_MATCH;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.dropDeltaTableWithRetry;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.getTablePropertyOnDelta;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.getTablePropertyOnTrino;
+import static io.trino.tests.product.utils.QueryExecutors.onDelta;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public class TestDeltaLakeTablePropertiesCompatibility
+        extends BaseTestDeltaLakeS3Storage
+{
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testTablePropertiesCaseSensitivity()
+    {
+        String tableName = "test_dl_table_properties_case_sensitivity_" + randomNameSuffix();
+        String tableDirectory = "databricks-compatibility-test-" + tableName;
+
+        onDelta().executeQuery(format("CREATE TABLE default.%s (col INT) USING DELTA LOCATION 's3://%s/%s' TBLPROPERTIES ('foo'='bar', 'Foo'='Baz')",
+                tableName,
+                bucketName,
+                tableDirectory));
+
+        try {
+            assertEquals(getTablePropertyOnTrino("default", tableName, "foo"), "bar");
```

Comment:
Instead of these lengthy assertions, it would be more straightforward to make sure that the rows selected from spark and from Trino coincide:

https://github.com/trinodb/trino/blob/73e0550074d9c1a7c04f4ac9fda28c627ecc5dbe/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeSelectCompatibility.java#L69-L84

Reply:
Indeed, being explicit about which properties we expect seems superior. Unfortunately, we don't really control which table properties the implementations will set. This makes it a bit more tricky to maintain and implement.

For instance,
```
Delta Lake OSS:
CREATE TABLE default.test (col INT) USING DELTA TBLPROPERTIES ('delta.minReaderVersion'='2', 'delta.minWriterVersion'='6', 'delta.columnMapping.mode'='id'); -- (2, 6) instead of (3, 7), since this is what I had at hand
SHOW TBLPROPERTIES default.test;
+-------------------------------+-----+
|key                            |value|
+-------------------------------+-----+
|delta.columnMapping.maxColumnId|1    |
|delta.columnMapping.mode       |id   |
|delta.minReaderVersion         |2    |
|delta.minWriterVersion         |6    |
+-------------------------------+-----+

---

Delta Lake Databricks:
CREATE TABLE default.test (col INT) USING DELTA TBLPROPERTIES ('delta.minReaderVersion'='2', 'delta.minWriterVersion'='6', 'delta.columnMapping.mode'='id');
SHOW TBLPROPERTIES default.test;
# | key | value |  
-- | -- | -- | --
1 | delta.checkpoint.writeStatsAsJson | false |  
2 | delta.checkpoint.writeStatsAsStruct | true |  
3 | delta.columnMapping.maxColumnId | 1 |  
4 | delta.columnMapping.mode | id |  
5 | delta.minReaderVersion | 2 |  
6 | delta.minWriterVersion | 6
```

Reply:
@jkylling what I'm hinting towards is to compare the results of the statements:

Trino
```
SELECT * FROM "test$properties";
```

Spark
```
SHOW TBLPROPERTIES default.test;
```

They should be the same for a given table right?